grid/chat/messages/maps/discord.suma

```diff
@@ -0,0 +1,253 @@
+
+  set if (input.afterTimestamp) {
+    afterId = call GetSnowflake(ts = input.afterTimestamp)
+  }
+
+  set if (input.beforeTimestamp) {
+    beforeId = call GetSnowflake(ts = input.beforeTimestamp)
+  }
+
+  http GET "/channels/{destination}/messages" {
+    request "application/json" {
+      query {
+        after = afterId
+        before = input.page || beforeId
+        limit = input.limit
+      }
+
+      headers {
+        "Authorization" = `Bot ${parameters.accessToken}`
+      }
+    }
+
+    response 200 "application/json" {
+      messages = call MapMessages(messages = body)
+      rateLimit = call MapRateLimitDetails(headers = headers)
+
+      map result {
+        nextPage = messages[messages.length - 1].id
+        messages = messages
+        rateLimit = rateLimit
+      }
+    }
+
+    response {
+      error = call MapDiscordError(
+        statusCode = statusCode,
+        error = body,
+        headers = headers,
+      )
+      
+      map error error
+    }
+  }
+}
+
+"""
+Operation MapMessages
+Returns array mapping messages in received body
+"""
+operation MapMessages {
+  listOfMessages = []
+
+  listOfMessages = args.messages.map(message => {
+    const resultMessage = {
+      id: message.id,
+      author: {
+        id: message.author.id,
+        username: message.author.username,
+      },
+      createdAt: Date.parse(message.timestamp),
+      text: message.content,
+    };
+
+    let reactions = [], attachments = [];
+
+    if (message.edited_timestamp) {
+      resultMessage.updatedAt = Date.parse(message.edited_timestamp);
+    }
+
+    if (message.thread) {
+      resultMessage.hasThread = true;
+      resultMessage.threadId = message.thread.id;	
+    }
+
+    if (message.reactions && message.reactions.length > 0) {
+      reactions = message.reactions.map(reaction => ({
+        emoji: reaction.emoji.name,
+        count: reaction.count,
+      }));
+
+      resultMessage.reactions = reactions
+    }
+
+    if (message.attachments && message.attachments.length > 0) {
+      const discordEpoch = 1420070400000;
+      const getTimestamp = (snowflake) => Number(BigInt(snowflake) >> BigInt(22)) + discordEpoch;
+
+      attachments = message.attachments.map(attachment => {		
+        const createdAt = getTimestamp(attachment.id);
+        const resultAttachment = {
+          id: attachment.id,
+          createdAt: createdAt,
+          fileName: attachment.filename,
+          url: attachment.url,
+        };
+
+        if (attachment.content_type) {
+          resultAttachment.mediaType = attachment.content_type;
+        }
+
+        return resultAttachment;
+      });
+
+      resultMessage.attachments = attachments;
+    }
+
+    if (message.message_reference && message.message_reference.message_id) {
+      resultMessage.parentId = message.message_reference.message_id;
+    }
+
+    resultMessage.url = `https://discord.com/api/channels/${message.channel_id}/messages/${message.id}`;
```

Comment:
Again, this should be a publicly accessible URL.

Reply:
Removed it

Reply:
Removed it